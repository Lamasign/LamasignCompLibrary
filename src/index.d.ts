
import { DefineComponent, Plugin } from "vue";

declare const vue3ComponentLibrary: Exclude<Plugin["install"], undefined>;

export default vue3ComponentLibrary;

export const BButton: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BBadge: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BAccordion: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BCardModal: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BCheckbox: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BDropdown: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BField: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BFileInput: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BInput: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BLink: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BMessage: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BNumberInput: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BRadio: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BSelect: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BSkeleton: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BSwitch: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BTable: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BTableColumn: Exclude<Plugin["install"], undefined> | DefineComponent;
export const Btag: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BToast: Exclude<Plugin["install"], undefined> | DefineComponent;
export const BTooltip: Exclude<Plugin["install"], undefined> | DefineComponent;