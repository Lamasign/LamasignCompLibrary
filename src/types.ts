import {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
} from "./components/accordion/BAccordion.types"


import {
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
} from './components/Badge/BBadge.types'

import {
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
} from './components/Button/BButton.types'

import {
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
} from './components/Checkbox/BCheckbox.types'

import {
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
} from './components/Dropdown/BDropdown.types'

import {
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap
} from './components/Field/BField.types'

import {
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum
} from './components/FileInput/BFileInput.types'

import {
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
} from './components/Input/BInput.types'

import {
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
} from './components/Link/BLink.types'

import {
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
} from './components/Message/BMessage.types'

import {
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
} from './components/NumberInput/BNumberInput.types'

import {
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
} from './components/Select/BSelect.types'

import {
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
} from './components/Skeleton/BSkeleton.types'

import {
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
} from './components/Switch/BSwitch.types'

import {
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
} from './components/Table/BTable.types'

import {
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
} from './components/Tag/BTag.types'

import {
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
} from './components/Toast/BToast.types'

import {
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
} from './components/Tooltip/BTooltip.types'

export {
	AccordionVariantsArray,
	AccordionVariantsEnum,
	AccordionVariantsMap,
	BadgeColorsArray,
	BadgeColorsEnum,
	BadgeColorsMap,
	BadgeSizeEnum,
	BadgeSizesArray,
	BadgeSizesMap,
	BadgeVariantsArray,
	BadgeVariantsEnum,
	BadgeVariantsMap,
	ButtonSizesEnum,
	ButtonVariantsEnum,
	ButtonTagsEnum,
	ButtonTypesEnum,
	ButtonSizesArray,
	ButtonSizesMap,
	ButtonVariantsArray,
	ButtonVariantsMap,
	ButtonTagsArray,
	ButtonTypesArray,
	CheckboxVariantsEnum,
	CheckboxVariantsArray,
	CheckboxsVariantsMap,
	DropdownVariantsEnum,
	DropdownDirectionEnum,
	DropdownDirectionMap,
	DropdownVariantsArray,
	DropdownVariantsMap,
	FIELD_STATUS_INJECTION_KEY,
	FieldStatusesArray,
	FieldStatusesEnum,
	FieldStatusesClassesMap,
	FieldStatusesTypes,
	FieldMessageStatusesClassesMap,
	InputFileStatusesArray,
	InputFileStatusesClassesMap,
	InputFileStatusesEnum,
	InputTypesEnum,
	InputTypesArray,
	InputStatusesArray,
	InputStatusesClassesMap,
	InputStatusesEnum,
	InputStatusesTypes,
	InputTypes,
	LinkDisabledClassesMap,
	LinkTags,
	LinkTagsArray,
	LinkTagsEnum,
	LinkVariantsArray,
	LinkVariantsEnum,
	LinkVariantsMap,
	MessageVariantsArray,
	MessageVariantsEnum,
	MessageVariantsMap,
	NumberInputStatusesArray,
	NumberInputStatusesClassesMap,
	NumberInputStatusesEnum,
	NumberInputStatusesTypes,
	SelectColorsEnum,
	SelectColorsArray,
	SelectColorsMap,
	SkeletonVariantsEnum,
	SkeletonVariantsArray,
	SkeletonVariantsMap,
	SkeletonAnimationsArray,
	SkeletonAnimationsEnum,
	SkeletonAnimationsMap,
	SkeletonColorsArray,
	SkeletonColorsEnum,
	SkeletonColorsMap,
	SwitchVariantsArray,
	SwitchVariantsEnum,
	SwitchVariantsMap,
	TableColumnHeaderVariantsMap,
	TableColumnVariantsArray,
	TableColumnVariantsEnum,
	TableColumnVariantsMap,
	TagVariantsArray,
	TagVariantsEnum,
	TagVariantsMap,
	TagsVariantsMap,
	ToastVariantsArray,
	ToastVariantsMap,
	ToastVariantsEnum,
	TooltipPositionEnum,
	TooltipPositionsArray,
	TooltipPositionsMap,
	TooltipVariantsArray,
	TooltipVariantsEnum,
	TooltipVariantsMap,
}
