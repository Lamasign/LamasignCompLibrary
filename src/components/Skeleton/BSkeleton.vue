<template>
  <component
    :is="tag"
    class="w-full h-full"
  >
    <div
      v-if="Variant === SkeletonVariantsEnum.RECTANGULAR"
      :class="finalClasses"
    />
    <div
      v-else-if="Variant === SkeletonVariantsEnum.CIRCULAR"
      :class="finalClasses"
    />
    <div
      v-else-if="Variant === SkeletonVariantsEnum.LINE"
      :class="finalClasses"
    />
    <div v-else-if="Variant === SkeletonVariantsEnum.TEXT">
      <div :class="finalClasses" />
      <div :class="finalClasses" />
      <div :class="finalClasses" />
    </div>
  </component>
</template>
<script lang='ts'>
import { computed, defineComponent, PropType } from '@vue/composition-api'
import { SkeletonColorsEnum, SkeletonVariantsEnum, SkeletonAnimationsEnum, SkeletonColorsMap, SkeletonVariantsMap, SkeletonAnimationsMap } from '@/components/Skeleton/BSkeleton.types'

export default defineComponent({
  name: 'BSkeleton',
  props: {
    Color: {
      type: String as PropType<SkeletonColorsEnum>,
      default: SkeletonColorsEnum.PLACEHOLDER,
    },
    Variant: {
      type: String as PropType<SkeletonVariantsEnum>,
      default: SkeletonVariantsEnum.LINE,
    },
    Animation: {
      type: String as PropType<SkeletonAnimationsEnum>,
      default: SkeletonAnimationsEnum.WAVE,
    },
    with: {
      type: Number,
      default: 40,
    },
    height: {
      type: Number,
      default: 40,
    },
    tag: {
      type: String,
      default: 'div',
    }
  },
  setup(props) {
    const colorClasses = computed(() => SkeletonColorsMap[props.Color])
    const variantClasses = computed(() => SkeletonVariantsMap[props.Variant])
    const animationClasses = computed(() => SkeletonAnimationsMap[props.Animation])

    const finalClasses = computed(() => `${colorClasses.value} ${variantClasses.value} ${animationClasses.value}`)

    return {
      SkeletonVariantsEnum,
      finalClasses,
    }
  },
})
</script>