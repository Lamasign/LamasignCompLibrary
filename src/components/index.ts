import BAccordion from './components/accordion/BAccordion.vue'
import BBadge from './components/Badge/BBadge.vue'
import BButton from './components/Button/BButton.vue'
import BCardModal from './components/CardModal/BCardModal.vue'
import BCheckbox from './components/Checkbox/BCheckbox.vue'
import BDropdown from './components/Dropdown/BDropdown.vue'
import BField from './components/Field/BField.vue'
import BFileInput from './components/FileInput/BFileInput.vue'
import BInput from './components/Input/BInput.vue'
import BLink from './components/Link/BLink.vue'
import BMessage from './components/Message/BMessage.vue'
import BNumberInput from './components/NumberInput/BNumberInput.vue'
import BRadio from './components/Radio/BRadio.vue'
import BSelect from './components/Select/BSelect.vue'
import BSkeleton from './components/Skeleton/BSkeleton.vue'
import BSwitch from './components/Switch/BSwitch.vue'
import BTable from './components/Table/BTable.vue'
import BTableColumn from './components/Table/BTableColumn.vue'
import BTag from './components/Tag/BTag.vue'
import BToast from './components/Toast/BToast.vue'
import BTooltip from './components/Tooltip/BTooltip.vue'

export {
	BAccordion,
	BBadge,
	BButton,
	BCardModal,
	BCheckbox,
	BDropdown,
	BField,
	BFileInput,
	BInput,
	BLink,
	BMessage,
	BNumberInput,
	BRadio,
	BSelect,
	BSkeleton,
	BSwitch,
	BTable,
	BTableColumn,
	BTag,
	BToast,
	BTooltip,
}
// export default components