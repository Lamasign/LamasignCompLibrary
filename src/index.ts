import { default as BAccordion } from './components/accordion/BAccordion.vue'
import { default as BBadge } from './components/Badge/BBadge.vue'
import { default as BButton } from './components/Button/BButton.vue'
import { default as BCardModal } from './components/CardModal/BCardModal.vue'
import { default as BCheckbox } from './components/Checkbox/BCheckbox.vue'
import { default as BDropdown } from './components/Dropdown/BDropdown.vue'
import { default as BField } from './components/Field/BField.vue'
import { default as BFileInput } from './components/FileInput/BFileInput.vue'
import { default as BInput } from './components/Input/BInput.vue'
import { default as BLink } from './components/Link/BLink.vue'
import { default as BMessage } from './components/Message/BMessage.vue'
import { default as BNumberInput } from './components/NumberInput/BNumberInput.vue'
import { default as BRadio } from './components/Radio/BRadio.vue'
import { default as BSelect } from './components/Select/BSelect.vue'
import { default as BSkeleton } from './components/Skeleton/BSkeleton.vue'
import { default as BSwitch } from './components/Switch/BSwitch.vue'
import { default as BTable } from './components/Table/BTable.vue'
import { default as BTableColumn } from './components/Table/BTableColumn.vue'
import { default as BTag } from './components/Tag/BTag.vue'
import { default as BToast } from './components/Toast/BToast.vue'
import { default as BTooltip } from './components/Tooltip/BTooltip.vue'

// const components = {
// 	BAccordion,
// 	BBadge,
// 	BButton,
// 	BCardModal,
// 	BCheckbox,
// 	BDropdown,
// 	BField,
// 	BFileInput,
// 	BInput,
// 	BLink,
// 	BMessage,
// 	BNumberInput,
// 	BRadio,
// 	BSelect,
// 	BSkeleton,
// 	BSwitch,
// 	BTable,
// 	BTableColumn,
// 	BTag,
// 	BToast,
// 	BTooltip,
// }
// export default components