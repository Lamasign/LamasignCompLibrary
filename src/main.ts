import Vue from 'vue'
import App from './App.vue'
import VueCompositionAPI from '@vue/composition-api'
import './assets/tailwind.css'
import * as components from './components'
Vue.config.productionTip = false

Vue.use(VueCompositionAPI)

new Vue({
	render: h => h(App),
}).$mount('#app')

function installLibrary() {

	Object.entries(components).forEach(([componentName, component]) => {
		Vue.component(componentName, component)
	})
}

Vue.use(installLibrary)

export default installLibrary
export * from './components'
export * from './utils'
