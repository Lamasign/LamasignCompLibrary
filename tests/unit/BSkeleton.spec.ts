import { createWrapper } from "./utils"
import BSkeleton from './components/Skeleton/BSkeleton.vue'
import { SkeletonAnimationsArray, SkeletonColorsArray, SkeletonVariantsArray } from '../../src/types'

describe('BSkeleton.vue', () => {
	it('renders correct variant based on variant props', () => {
		expect.assertions(SkeletonVariantsArray.length + 1)
		const defaultWrapper = createWrapper(BSkeleton)

		expect(defaultWrapper.element).toMatchSnapshot()

		SkeletonVariantsArray.forEach(variant => {
			const wrapperWithProps = createWrapper(BSkeleton, {
				propsData: {
					variant
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
		})
	})

	it('renders correct variant based on variant props', () => {
		expect.assertions(SkeletonColorsArray.length + 1)
		const defaultWrapper = createWrapper(BSkeleton)

		expect(defaultWrapper.element).toMatchSnapshot()

		SkeletonColorsArray.forEach(Color => {
			const wrapperWithProps = createWrapper(BSkeleton, {
				propsData: {
					Color
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
		})
	})

	it('renders correct variant based on variant props', () => {
		expect.assertions(SkeletonAnimationsArray.length + 1)
		const defaultWrapper = createWrapper(BSkeleton)

		expect(defaultWrapper.element).toMatchSnapshot()

		SkeletonAnimationsArray.forEach(Animation => {
			const wrapperWithProps = createWrapper(BSkeleton, {
				propsData: {
					Animation
				}
			})
			expect(wrapperWithProps.element).toMatchSnapshot()
		})
	})


})