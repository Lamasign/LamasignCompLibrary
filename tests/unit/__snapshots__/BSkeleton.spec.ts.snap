// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BSkeleton.vue renders correct variant based on variant props 1`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 2`] = `
<div
  class="w-full h-full"
>
  <div>
    <div
      class="bg-gray-300 leading-relaxed mb-3 h-3 animation-wave"
    />
     
    <div
      class="bg-gray-300 leading-relaxed mb-3 h-3 animation-wave"
    />
     
    <div
      class="bg-gray-300 leading-relaxed mb-3 h-3 animation-wave"
    />
  </div>
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 3`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed rounded-full mb-3 w-24 h-24 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 4`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 lg:h-48 leading-relaxed animate-pulse md:h-36 w-full object-cover object-center animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 5`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 6`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 7`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-blue-light leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 8`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-orange-light leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 9`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-red-light leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 10`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-green-light leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 11`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 12`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 13`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-wave"
  />
</div>
`;

exports[`BSkeleton.vue renders correct variant based on variant props 14`] = `
<div
  class="w-full h-full"
>
  <div
    class="bg-gray-300 leading-relaxed mb-3 w-full h-3 animate-pulse animation-pulse"
  />
</div>
`;
